crates/runtime/src/memory.rs

```diff
@@ -565,6 +565,7 @@ impl SharedMemory {
     /// Implementation of `memory.atomic.notify` for this shared memory.
     pub fn atomic_notify(&self, addr_index: u64, count: u32) -> Result<u32, Trap> {
         validate_atomic_addr(&self.0.def.0, addr_index, 4, 4)?;
+        log::trace!("memory.atomic.notify(addr={addr_index}, count={count})");
```

Comment:
instead of using the format string, use `log::trace!("memory.atomic.notify", addr=addr_index, count=count);` to provide structured values.

Reply:
I tried that but it complains: I think to use the key-value syntax we need to set `target: ...,` up front and still provide a string at the end (see [`log::log!`](https://docs.rs/log/latest/log/macro.log.html))?

Reply:
oh, my bad. I just realized this is `log` and the key value syntax only works with the `tracing` crate, and I forgot we use `log` in wasmtime-runtime in order to keep the dependencies as small as possible, but `tracing` in wasmtime-wit-bindgen, wasi, etc.

So, your PR is good to go as is. I don't think we need should using `log` vs `tracing` down in the runtime now, unless someone really wants to.